### PR TITLE
Wallet API: functions for supporting/creating view only wallets

### PR DESCRIPTION
**Pull request by Jaqueeee**

### CODE REVIEW

File: src/wallet/api/wallet.cpp [hunk 1]
```diff
@@ -277,6 +277,49 @@ bool WalletImpl::create(const std::string &path, const std::string &password, co
     return true;
 }
 
+bool WalletImpl::createWatchOnly(const std::string &path, const std::string &password, const std::string &language) const
+{
+    clearStatus();
+    tools::wallet2 * view_wallet = new tools::wallet2(m_wallet->testnet());
```
Comment (vtnerd) [inline src/wallet/api/wallet.cpp]: I think some smart pointers could be helpful in here, this can be `unique_ptr`.

Comment (Jaqueeee) [inline src/wallet/api/wallet.cpp]: Good idea. Changed to `unique_ptr`.

**Comment by Jaqueeee** [comment]
@vtnerd 
Yes, a view only wallet can actually be created from a view only wallet. But i've disabled the button in GUI to avoid confusion.

**Comment by fluffypony** [comment]
Plz rebase

**Comment by Jaqueeee** [comment]
Rebased

**Pull request merged**